overviewer_core/textures.py

```diff
@@ -830,6 +862,331 @@ def generate_texture_tuple(self, img):
             return None
         return (img, self.generate_opaque_mask(img))
 
+    #
+    # This part if for reading the models/*.json assets. 
+    #
+
+    models = {}
+
+    def load_model(self, modelname):
+        if modelname in self.models:
+            return self.models[modelname]
+
+        fileobj = self.find_file('assets/minecraft/models/' + modelname +
+                                 '.json', verbose=logging.getLogger().isEnabledFor(logging.DEBUG))
+        self.models[modelname] = json.load(fileobj)
+        fileobj.close()
+
+        if 'parent' in self.models[modelname]:
+            parent = self.load_model(re.sub('.*:', '', self.models[modelname]['parent']))
+            if 'textures' in parent:
+                self.models[modelname]['textures'].update(parent['textures'])
+            if 'elements' in parent:
+                if 'elements' in self.models[modelname]:
+                    self.models[modelname]['elements'] += parent['elements']
+                else:
+                    self.models[modelname]['elements'] = parent['elements']
+            del self.models[modelname]['parent']
+
+        self.models[modelname] = self.normalize_model(modelname)
+        return self.models[modelname]
+
+    # fix known inconsistencies in model info
+    def normalize_model(self, modelname):
+        if modelname == 'block/observer':
+            self.models[modelname] = deepcopy(self.models[modelname])
+            self.models[modelname]['elements'][0]['faces']['up']['uv'] = [0, 0, 16, 16]
+        elif modelname == 'block/loom':
+            self.models[modelname] = deepcopy(self.models[modelname])
+            self.models[modelname]['elements'][0]['faces']['up']['texturerotation'] = 180
+            self.models[modelname]['elements'][0]['faces']['down']['texturerotation'] = 180
+        elif modelname == 'block/barrel' or modelname == 'block/barrel_open':
+            self.models[modelname] = deepcopy(self.models[modelname])
+            self.models[modelname]['elements'][0]['faces']['north']['texture'] = '#up'
+            self.models[modelname]['elements'][0]['faces']['south']['texture'] = '#down'
+            self.models[modelname]['elements'][0]['faces']['down']['texture'] = '#north'
+            self.models[modelname]['elements'][0]['faces']['up']['texture'] = '#south'
+            self.models[modelname]['elements'][0]['faces']['east']['texturerotation'] = 90
+            self.models[modelname]['elements'][0]['faces']['west']['texturerotation'] = 90
+        elif modelname == 'block/dropper_vertical' or modelname == 'block/dispenser_vertical':
+            self.models[modelname] = deepcopy(self.models[modelname])
+            self.models[modelname]['elements'][0]['faces']['north']['texture'] = '#up'
+            self.models[modelname]['elements'][0]['faces']['up']['texture'] = '#north'
+
+        return self.models[modelname]
+
+    def build_block_from_model(self, modelname, blockstate={}):
```

Comment:
FYI there's an recursion error creating textures for worlds created after 1.19.3 that's being caused by the new experimental [chiseled bookshelf](https://minecraft.fandom.com/wiki/Chiseled_Bookshelf) models added for 1.20. 

I haven't had time to look into this more deeply, but for now this suggestion will skip building the problematic models:

```suggestion
    def build_block_from_model(self, modelname, blockstate={}):
        if 'chiseled_bookshelf' in modelname:
            return
```